src/cargo/ops/cargo_install.rs

```diff
@@ -293,7 +295,7 @@ fn select_pkg<'a, T>(mut source: T,
                 }
                 None => None,
             };
-            let vers = vers.as_ref().map(|s| &**s);
+            let vers = vers.as_ref().map(|s| &**s).or(range);
```

Comment:
To avoid passing both a version and a range through the whole process, I decided to pass the range if the version is `None` to `Dependency::parse_no_deprecated`. I'm not sure if it is the right thing to do?

Does `Dependency::parse_no_deprecated` accept a version range by design, or is that the mistake that allows `--vers` to accept ranges? Can I rely on that behaviour?